### PR TITLE
Move link-checker to github actions

### PR DESCRIPTION
**Pull request by c-martinez**
- [X] I followed the [CONTRIBUTING guidelines](../blob/master/CONTRIBUTING.md).

Below, describe what this Pull Request adds:
This PR adds a github action for checking broken links, removing the need to do this on Travis.

### CODE REVIEW

**Review by bouweandela — CHANGES_REQUESTED** [review]
To *move* the link checker to GitHub actions, I would expect the travis tests would need to be removed as well? I see the action is not running yet, but I guess that is because it is not in `master` yet?

**Comment by c-martinez** [comment]
Good point. I've removed the the travis files. Indeed action is not yet running because this is not yet merged, but you can see the action running on [my fork](https://github.com/c-martinez/guide/actions?query=workflow%3A%22Link+Checker%22)

**Comment by bouweandela** [comment]
Great! Would it be possible to display only broken links? At the moment a lot of scrolling through the output is needed to find which links are broken.

File: .github/workflows/link-checker.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            broken_max=10
+            broken=$(grep ERROR ./link-checker/out.md | wc -l)
+
+            if [ "$broken" -gt "$broken_max" ]; then
+                echo "Number of broken links (${broken}) exceeds maximum allowed number (${broken_max})."
+                exit 1
+            else
+                echo "Number of broken links (${broken}) less than or equal to maximum allowed number (${broken_max})."
+                exit 0
+            fi
```
Comment (bouweandela) [inline .github/workflows/link-checker.yml:26]: Maybe you could also display the broken links here?

**Comment by bouweandela** [comment]
There was also a check on broken links in changed files only, specifically for pull requests. Would it be possible to add that to GitHub actions too?

**Comment by c-martinez** [comment]
Good point -- I've added an additional action for checking on pull requests.

I also noticed that link-checker action is deprecated in favour of (lychee-action)[https://github.com/lycheeverse/lychee-action], so this will work for the moment, but we should consider upgrading in the future.

**Review by bouweandela — APPROVED** [review]

**Pull request merged**